gateway/ln-gateway/src/lib.rs

```diff
@@ -537,105 +578,149 @@ impl Gateway {
-                        action: Some(Action::Cancel(Cancel {
-                            reason: "Insufficient Liquidity".to_string(),
-                        })),
-                        payment_hash: htlc_request.payment_hash,
-                        incoming_chan_id: htlc_request.incoming_chan_id,
-                        htlc_id: htlc_request.htlc_id,
-                    };
+                error!("Error relaying incoming HTLC: {error:?}");
+
+                let outcome = InterceptHtlcResponse {
+                    action: Some(Action::Cancel(Cancel {
+                        reason: "Insufficient Liquidity".to_string(),
+                    })),
+                    payment_hash: payment_hash.to_bytes(),
+                    incoming_chan_id: htlc_request.incoming_chan_id,
+                    htlc_id: htlc_request.htlc_id,
+                };
 
-                    if let Err(error) = lightning_context.lnrpc.complete_htlc(outcome).await {
-                        error!("Error sending HTLC response to lightning node: {error:?}");
-                    }
+                if let Err(error) = lightning_context.lnrpc.complete_htlc(outcome).await {
+                    error!("Error sending HTLC response to lightning node: {error:?}");
                 }
-
-                continue;
             }
 
-            // Check if the HTLC corresponds to a federation supporting legacy Lightning.
-            if let Some(short_channel_id) = htlc_request.short_channel_id {
-                // Just forward the HTLC if we do not have a federation that
-                // corresponds to the short channel id
-                if let Some(client) = self
-                    .federation_manager
-                    .read()
-                    .await
-                    .get_client_for_scid(short_channel_id)
-                {
-                    let cf = client
-                        .borrow()
-                        .with(|client| async {
-                            let htlc = htlc_request.clone().try_into();
-                            if let Ok(htlc) = htlc {
-                                match client
-                                    .get_first_module::<GatewayClientModule>()
-                                    .gateway_handle_intercepted_htlc(htlc)
-                                    .await
-                                {
-                                    Ok(_) => return Some(ControlFlow::<(), ()>::Continue(())),
-                                    Err(e) => {
-                                        error!("Got error intercepting HTLC: {e:?}, will retry...");
-                                    }
+            return true;
+        }
+
+        false
+    }
+
+    /// Tries to handle an HTLC using the legacy lightning protocol.
+    /// Returns `true` if the HTLC was handled, `false` otherwise.
+    async fn try_handle_htlc_ln_legacy(&self, htlc_request: &InterceptHtlcRequest) -> bool {
```

Comment:
Same comment as above, I think we can use a `Result` here

Reply:
Done.